test/acceptance/perf/setup/consul-server.tf

```diff
@@ -0,0 +1,244 @@
+  secret_string = random_id.gossip_encryption_key.b64_std
+}
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
+resource "tls_private_key" "ssh" {
+  algorithm = "RSA"
+}
+
+resource "aws_key_pair" "key_pair" {
+  key_name   = "${local.name}-${random_string.secret_suffix.result}"
+  public_key = tls_private_key.ssh.public_key_openssh
+}
+
+resource "null_resource" "save_key" {
+  triggers = {
+    key = tls_private_key.ssh.private_key_pem
+  }
+
+  provisioner "local-exec" {
+    command = <<EOF
+      mkdir -p ${path.module}/.ssh
+      echo "${tls_private_key.ssh.private_key_pem}" > ${path.module}/.ssh/id_rsa
+      chmod 0600 ${path.module}/.ssh/id_rsa
+EOF
+  }
+}
+
+# Create an IAM policy for allowing instances that are running
+# Consul agent can use to list the consul servers.
+resource "aws_iam_policy" "consul_retry_join" {
+  name        = "${local.name}-consul-retry-join"
+  description = "Allows Consul nodes to describe instances for joining."
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": "ec2:DescribeInstances",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+# Create an IAM role for the auto-join
+resource "aws_iam_role" "consul_retry_join" {
+  name = "${local.name}-consul-retry-join"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+# Attach the policy
+resource "aws_iam_role_policy_attachment" "consul_retry_join" {
+  role       = aws_iam_role.consul_retry_join.name
+  policy_arn = aws_iam_policy.consul_retry_join.arn
+}
+
+# Create the instance profile
+resource "aws_iam_instance_profile" "consul_retry_join" {
+  name = "${local.name}-consul-retry-join"
+  role = aws_iam_role.consul_retry_join.name
+}
+
+resource "random_uuid" "bootstrap_token" {}
+
+resource "aws_secretsmanager_secret" "bootstrap_token" {
+  name                    = "${local.name}-bootstrap-token-${random_string.secret_suffix.result}"
+  recovery_window_in_days = 0
+}
+
+resource "aws_secretsmanager_secret_version" "bootstrap_token" {
+  secret_id     = aws_secretsmanager_secret.bootstrap_token.id
+  secret_string = random_uuid.bootstrap_token.result
+}
+
+module "consul-server" {
+  // https://github.com/hashicorp/consul-global-scale-benchmark doesn't support ACLs so I forked it
+  source               = "github.com/erichaberkorn/consul-global-scale-benchmark/infrastructure/consul-server-ec2"
+  project              = local.name
+  vpc_id               = module.vpc.vpc_id
+  subnet_ids           = module.vpc.public_subnets
```

Comment:
Since this is on public subnets, are the servers exposed to the public internet? And if so, are they restricted from the public internet somehow (by security group / ingress ip or etc)?

Just double checking that the Consul servers aren't subject to any extra/unknown load from "random hackers" trying to get into our EC2 instances.

Reply:
I moved the Consul servers to a private subnet